L1Trigger/L1THGCal/python/customClustering.py

```diff
@@ -88,6 +88,55 @@ def custom_3dclustering_histoMax(process,
     return process
 
 
+def custom_3dclustering_histoInterpolatedMax1stOrder(process,
+        distance = 0.01,
+        nBins_R = 36,
+        nBins_Phi = 216,
+        binSumsHisto = cms.vuint32(13,               #0
+                                   11, 11, 11,       # 1 - 3
+                                   9, 9, 9,          # 4 - 6
+                                   7, 7, 7, 7, 7, 7, # 7 - 12
+                                   5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5,  # 13 - 27
+                                   3, 3, 3, 3, 3, 3, 3, 3  # 28 - 35
+                                   ),
+        neighbour_weights=cms.vdouble(  0    , 0.25, 0   ,
+                                        0.25 , 0  ,  0.25,
+                                        0    , 0.25, 0
+                                     )
+                                           
+
+        ):
+    process = custom_3dclustering_histoMax( process, distance, nBins_R, nBins_Phi, binSumsHisto )    
+    parameters_c3d = process.hgcalBackEndLayer2Producer.ProcessorParameters.C3d_parameters
+    parameters_c3d.type_multicluster = cms.string('HistoInterpolatedMaxC3d')
+    parameters_c3d.neighbour_weights = neighbour_weights
+    return process
+
+def custom_3dclustering_histoInterpolatedMax2ndOrder(process,
```

Comment:
Since the `1stOrder` and `2ndOrder` functions are exactly the same except for the default parameter values, it could be merged into one single function `custom_3dclustering_histoInterpolatedMax` and have the `1stOrder` and `2ndOrder` functions that just call this one with different `neighbour_weights` (which in that case doesn't need to be passed as parameter of the  `1stOrder` and `2ndOrder` functions)